src/Sylius/Bundle/OrderBundle/Controller/OrderController.php

```diff
@@ -183,6 +180,73 @@ public function clearAction(Request $request)
     }
 
     /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function completeAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::UPDATE);
+        $resource = $this->findOr404($configuration);
+
+        $form = $this->resourceFormFactory->create($configuration, $resource);
+
+        if (in_array($request->getMethod(), ['POST', 'PUT', 'PATCH'], true) && $form->handleRequest($request)->isValid()) {
+            $resource = $form->getData();
+
+            $event = $this->eventDispatcher->dispatchPreEvent(ResourceActions::UPDATE, $configuration, $resource);
+
+            if ($event->isStopped() && !$configuration->isHtmlRequest()) {
+                throw new HttpException($event->getErrorCode(), $event->getMessage());
+            }
+            if ($event->isStopped()) {
+                $this->flashHelper->addFlashFromEvent($configuration, $event);
+
+                return $this->redirectHandler->redirectToResource($configuration, $resource);
+            }
+
+            if ($configuration->hasStateMachine()) {
+                $this->stateMachine->apply($configuration, $resource);
```

Comment:
Of course, I could do that, but if we leave that as it is now, we could define e.q. state machine or redirect in the same way as in another checkout steps. I am afraid that customizing this method and changing routing could become unclear.